controller/lib/core/controller.h

```diff
@@ -52,6 +52,10 @@ class Controller {
   // objects.
   std::optional<FlowIntegrator> flow_integrator_ = FlowIntegrator();
   std::optional<FlowIntegrator> uncorrected_flow_integrator_ = FlowIntegrator();
+
+  // This state allows resetting integrators when transitioning from Off state
+  // to On state.
```

Comment:
The sentence about what this is used for is helpful, but also worthwhile saying what the variable *is*, too.  Like, "Whether or not the ventilator was off after the last call to Run()."

Also nit, maybe invert the boolean, ventilator_was_on_?  Dunno about you but I find positive booleans easier to reason about.

Reply:
Seems like it is worth adding indeed, I'm quite agnostic about the logic inversion myself, but if it helps some people, I will gladly change it!